knx/dpt/formats.go

```diff
@@ -30,6 +30,40 @@ func unpackB1(data []byte, b *bool) error {
 	return nil
 }
 
+func packB4(b0 bool, b1 bool, b2 bool, b3 bool) byte {
+	var b int
+	b = 0
+	if b3 {
+		b += 1 << 0
+	}
+	if b2 {
+		b += 1 << 1
+	}
+	if b1 {
+		b += 1 << 2
+	}
+	if b0 {
+		b += 1 << 3
+	}
```

Comment:
Use `|=` operator instead.

Reply:
It is a bit odd that `b0` is at index 3, and `b3` and index 0. The naming is a little confusion. In the KNX standard, typically the least significant bit is `b0`.

Reply:
Hm, have to think about this. I did this because reading RGBW from left to right and wanted to be red = 0

Reply:
I changed the direction. Now the direction of the colors looks odd :)

```go
validBits := packB4([4]bool{d.WhiteValid, d.BlueValid, d.GreenValid, d.RedValid})
```

Reply:
Either way there is going to be some weirdness. Best to stick to the standard in that case.